src/pages/projectSettings/tabs/NotificationsTab/SubscriptionField.tsx

```diff
@@ -0,0 +1,93 @@
+import styled from "@emotion/styled";
+import ExpandableCard from "@leafygreen-ui/expandable-card";
+import { Body } from "@leafygreen-ui/typography";
+
+import { Field } from "@rjsf/core";
+import { size } from "constants/tokens";
+import { triggers } from "constants/triggers";
+import { Subscriber } from "gql/generated/types";
+import { ResourceType, Trigger } from "types/triggers";
+import { toSentenceCase } from "utils/string";
+import { SubscriptionSelect } from "./NotificationsRow";
+
+export const getSubscriberType = (subscriber: Subscriber) => {
+  if (subscriber === undefined) {
+    return "";
+  }
+  let key: string = "";
+  Object.keys(subscriber).forEach((k) => {
+    if (subscriber[k] !== null) {
+      key = k;
+    }
+  });
+  return key;
+};
+
+export const getTrigger = (trigger: string, resourceType: string) => {
+  let triggerData: Trigger = {
+    trigger: "",
+    resourceType: ResourceType.VERSION,
+    regexSelectors: [],
+    label: "",
+  };
+  triggers.forEach((t) => {
+    if (t.trigger === trigger && t.resourceType === resourceType) {
+      triggerData = t;
+      return t;
```

Comment:
I don't think there's any reason to overwrite `triggerData` here; you can just return `t`!

Reply:
I need to do it to avoid returning void type, because that is harder to work with